csrc/ir/utils.cpp

```diff
@@ -1103,6 +1103,31 @@ int64_t getVectorizeSize(const TensorView* tv) {
   return 1;
 }
 
+bool hasTrivialAllocationDomain(const TensorView* tv) {
+  if (!tv->hasAllocation()) {
+    return true;
+  }
+  const std::vector<IterDomain*>& alloc = tv->getMaybeAllocationDomain();
+  const std::vector<IterDomain*>& rf = tv->getMaybeRFactorDomain();
```

Comment:
Is it possible to get the allocation/rfactor domains, then remove the broadcasts and reductions.
Then we can check if the filter allocation domain and the RFactor domains are permutations or not?

Reply:
In this case, we do not just need it to be a permutation, but actually identical. That's why I am checking sameAs in the loop below while removing broadcast and reduction domains. I could potentially use `TensorDomain::noReductions()` and `TensorDomain::noBroadcasts()` first, but I would still need to loop over the two vectors to call `sameAs`, so I just avoided building the temporaries and fused those loops here.

Reply:
In this case do we need to call `sameAs` or will checking equality of pointers be enough?

Reply:
If you're not against temporaries, we can remove the loop by first filtering with `noReductions` and `noBroadcasts`, then comparing the filter vectors with std::equal, where `sameAs` can be used as part of lambda.

Reply:
`sameAs` is probably not needed; after all we commonly use `std::is_permutation` to check that allocation domain is just a stride ordering. I'll change to use the helper functions (with temporaries) because I think it'll make it clearer.